### PR TITLE
Upgrade to Micronaut Framework 4.0.0 GA.

### PR DESCRIPTION
**Pull request by jeremyg484**
## What's changed?
Update all relevant versions in recipes and tests for Micronaut Framework 4.0.0

Note - Version 4.0.0 of all of the Micronaut modules have been published, but the platform bom and the build plugins 
still need to be published before these tests will pass. Once those artifacts are available and all of the tests are 
once again green, then we should be clear to merge this and publish a patch release.

### Checklist
- [x] I've added unit tests to cover both positive and negative cases

### CODE REVIEW

File: src/test/java/org/openrewrite/java/micronaut/UpdateBuildToMicronaut4VersionTest.java [hunk 1]
```diff
@@ -35,12 +35,12 @@ public void defaults(RecipeSpec spec) {
...
     @Language("properties")
     private final String initialGradleProperties = """
-            micronautVersion=3.9.0    
+            micronautVersion=3.9.0
         """;
 
     @Language("properties")
     private final String v4GradleProperties = """
-            micronautVersion=4.0.0-RC1    
+            micronautVersion=4.0.
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/micronaut/UpdateBuildToMicronaut4VersionTest.java:43]:
```suggestion
            micronautVersion=4.0.0
```

**Review by timtebeek — APPROVED** [review]

**Comment by timtebeek** [comment]
So if I'm correct we're waiting for 4.0.0 to show here: https://repo1.maven.org/maven2/io/micronaut/micronaut-bom/4.0.0
If anyone spots that before I do feel free to ping me either here or on Slack, such that we can get this one through & released.

**Comment by jeremyg484** [comment]
> So if I'm correct we're waiting for 4.0.0 to show here: https://repo1.maven.org/maven2/io/micronaut/micronaut-bom/4.0.0 If anyone spots that before I do feel free to ping me either here or on Slack, such that we can get this one through & released.

Note, things have been reorganized a bit for 4.0.0, so we're actually waiting on:

https://repo1.maven.org/maven2/io/micronaut/platform/micronaut-parent/4.0.0/

and

https://repo1.maven.org/maven2/io/micronaut/platform/micronaut-platform/4.0.0/

I'll ping you as soon as they're up and I've got these tests passing. (The Gradle and Maven build plugins that were also necessary have now been published, so this is the final step.)

**Comment by jeremyg484** [comment]
@timtebeek @sambsnyd With this merged successfully, we are good to release version 2.1.0 of this module now whenever you guys are available to do so.

**Comment by timtebeek** [comment]
Done! https://github.com/openrewrite/rewrite-micronaut/releases/tag/v2.1.0
As you probably already knew, but still :)

Congrats on the 4.0 release!

**Pull request merged**